examples/QIR/Simulation/README.md

```diff
@@ -0,0 +1,193 @@
+
+- the driver `IRuntimeDriver` : this is critical for the Runtime to operate, implementing qubit allocation and other backend-specific functionality
+- the gate set `IQuantumGateSet` : only used by those backends that want to provide the Q# instruction set
+- the diagnostics `IDiagnostics` : optional but helpful component to provide debug information of the backend state
+
+The first component of the Runtime stack is the QIR Bridge located at [lib/QIR/bridge-rt.ll](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QIR/bridge-rt.ll) which translates between QIR and the Runtime implementation.
+Different parts of the QIR spec are then implemented in C++ in the [lib/QIR](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/lib/QIR) folder.
+For example, the `__quantum__rt__string_create` QIR function is translated to the C++ `quantum__rt__string_create` function which resides in [lib/QIR/strings.cpp](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QIR/strings.cpp).
+For others such as `__quantum__rt__qubit_allocate`, the Runtime will eventually call the implementation provided by the driver.
+
+Other components that are used by QIR simulators for Q# are provided by [lib/QSharpCore](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/lib/QSharpCore) and [lib/QSharpFoundation](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/lib/QSharpFoundation).
+In particular, the `QSharpCore` component provides the Q# instruction set.
+Similar to above, a bridge [lib/QSharpCore/qsharp-core-qis.ll](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll) first translates a function such as `__quantum__qis__h__body` to `quantum__qis__h__body` located in [lib/QSharpCore/intrinsics.cpp](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QSharpCore/intrinsics.cpp), which then calls upon the specific implementation provided by the gate set interface.
+
+A qubit manager (see next section) is also provided in [lib/QIR/QubitManager.cpp](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QIR/QubitManager.cpp).
+
+## Qubit Management
+
+Generally, we can distinguish three layers of qubit representation or abstractions.
+Two of those are frequently encountered in quantum computing literature, the *logical qubit* and *the physical* qubit.
+
+A logical qubit is an idealized entity that represents the Qubit from quantum computation theory: a perfect two-level system that behaves as expected when applying quantum gates or measurements.
+The physical qubit on the other hand is a real physical system that *approximately* behaves as a two-level system, and is imperfectly manipulated and isolated from the environment, leading to a limited reliability and information life-time.
```

Comment:
I'm not sure that I agree with this definition of "logical" and "physical qubits"? Normally, I'd think of a logical qubit as being a two-level degree of freedom in a decoherence free subsystem, error correcting code, or other encoding into a larger register of physical qubits. Similarly, excepting errors such as leakage and coherent memory in the environment, physical qubits can be true two-level systems while still experiencing noise.

Reply:
I'm not quite sure how our descriptions diverge:
- "a two-level degree of freedom in a decoherence free subsystem" to me sounds equivalent to "a perfect two-level system that behaves as expected [in reference to the theory, i.e. without noise/decoherence]", but maybe this could be phrased better.
- "physical qubits can be true two-level systems while still experiencing noise" I guess here you are getting at the possibility of a perfect two-level physical qubit, but as far as I know, for practical qubit systems the "two-levelness" is achieved by sufficiently separating two energy eigenstates from the rest of the spectrum. I could be wrong on this though, so please feel free to correct me on this.